common/kvstore/tablestore/table_view.go

```diff
@@ -5,14 +5,17 @@ import (
 	"github.com/Layr-Labs/eigenda/common/kvstore"
 	"github.com/syndtr/goleveldb/leveldb/iterator"
 	"github.com/syndtr/goleveldb/leveldb/util"
+	"time"
 )
 
 var _ kvstore.Table = &tableView{}
 
 // tableView allows data in a table to be accessed as if it were a Store.
```

Comment:
Table view tries to appear like a store, by wrapping two stores...these are getting too complicated and won't be simple to use.

Reply:
This isn't actually wrapping two stores, this class has access to two different interfaces on the same store. This constructor isn't publicly exported, so it's impossible for an end user to instantiate this class with two different stores.

That being said, in order to simplify this method, I've modified the function signature to make things easier to read. 
The reason why this class needed access to the parent `TableStore` class was for the methods `Shutdown()`, `Destroy()`, and `NewBatch()`. Instead of passing in the whole `TableStore`, we can just pass in lambdas for the functionality needed by the table view.

```
func newTableView(
	base kvstore.Store,
	name string,
	prefix uint32,
	shutdown func() error,
	destroy func() error,
	batchBuilder func() kvstore.TableStoreBatch) kvstore.Table {
```

Reply:
The specific issue here is that `tableView` doesn't appear to be Store, it's trying to be a `Table` (e.g. the TTL related interfaces do not exist for Store)

Reply:
The interface `Table` currently extends the interface `Store`. A user can interact with a table as if it were a regular `Store` object.

```
type Table interface {
	Store

	Name() string
	TableKey(key []byte) TableKey
	PutWithTTL(key []byte, value []byte, ttl time.Duration) error
	PutWithExpiration(key []byte, value []byte, expiryTime time.Time) error
	NewTTLBatch() TTLStoreBatch
}
```